Tests/SwiftSyntaxMacroExpansionTest/MacroReplacementTests.swift

```diff
@@ -113,4 +114,230 @@ final class MacroReplacementTests: XCTestCase {
+    let macro: DeclSyntax =
+      """
+      macro gen(a: Array<Int>) = #otherMacro(first: a)
+      """
+
+    let use: ExprSyntax =
+      """
+      #gen(a: [1, 2, 3])
+      """
+
+    let macroDecl = macro.as(MacroDeclSyntax.self)!
+    let definition = try macroDecl.checkDefinition()
+    guard case let .expansion(expansion, replacements, genericReplacements) = definition else {
+      XCTFail("not a normal expansion")
+      return
+    }
+
+    XCTAssertEqual(genericReplacements.count, 0)
+
+    let expandedSyntax = macroDecl.expand(
+      use.as(MacroExpansionExprSyntax.self)!,
+      definition: expansion,
+      replacements: replacements,
+      genericReplacements: genericReplacements
+    )
+    assertStringsEqualWithDiff(
+      expandedSyntax.description,
+      """
+      #otherMacro(first: [1, 2, 3])
+      """
+    )
+  }
+
+  func testMacroGenericArgumentExpansion_replaceInner() throws {
+    let macro: DeclSyntax =
+      """
+      macro gen<A>(a: Array<A>) = #reduce<A>(first: a)
+      """
+
+    let use: ExprSyntax =
+      """
+      #gen<Int>(a: [1, 2, 3])
+      """
+
+    let macroDecl = macro.as(MacroDeclSyntax.self)!
+    let definition = try macroDecl.checkDefinition()
+    guard case let .expansion(expansion, replacements, genericReplacements) = definition else {
+      XCTFail("not a normal expansion")
+      return
+    }
+
+    XCTAssertEqual(genericReplacements.count, 1)
+
+    let expandedSyntax = macroDecl.expand(
+      use.as(MacroExpansionExprSyntax.self)!,
+      definition: expansion,
+      replacements: replacements,
+      genericReplacements: genericReplacements
+    )
+    assertStringsEqualWithDiff(
+      expandedSyntax.description,
+      """
+      #reduce<Int>(first: [1, 2, 3])
+      """
+    )
+  }
+
+  func testMacroGenericArgumentExpansion_array() throws {
+    let macro: DeclSyntax =
+      """
+      macro gen(a: Array<Int>) = #other<A>(first: a)
+      """
+
+    let use: ExprSyntax =
+      """
+      #otheren<Int>(a: [1, 2, 3])
+      """
+
+    let macroDecl = macro.as(MacroDeclSyntax.self)!
+    let definition = try macroDecl.checkDefinition()
+    guard case let .expansion(expansion, replacements, genericReplacements) = definition else {
+      XCTFail("not a normal expansion")
+      return
+    }
+
+    XCTAssertEqual(genericReplacements.count, 0)
+
+    let expandedSyntax = macroDecl.expand(
+      use.as(MacroExpansionExprSyntax.self)!,
+      definition: expansion,
+      replacements: replacements,
+      genericReplacements: genericReplacements
+    )
+    assertStringsEqualWithDiff(
+      expandedSyntax.description,
+      """
+      #other<A>(first: [1, 2, 3])
+      """
+    )
+  }
+
+  func testMacroExpansion_dontCrashOnDuplicates() throws {
+    let macro: DeclSyntax =
+      """
+      macro gen(a: Array<Int>) = #other<A>(first: a)
+      """
+
+    let use: ExprSyntax =
+      """
+      #otheren<Int>(a: [1, 2, 3])
```

Comment:
Is this a typo and should this be the following?

```suggestion
      #other<Int>(a: [1, 2, 3])
```

Reply:
typo oh no, guess the test run is going to fail on this now 😮‍💨 

Gotta fix other things anyway, thanks